addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -476,8 +476,8 @@ def _get_l10n_in_edi_ewaybill_line_details(self, line, line_tax_details, sign):
             "qtyUnit": line.product_id.uom_id.l10n_in_code and line.product_id.uom_id.l10n_in_code.split("-")[0] or "OTH",
             "taxableAmount": self._l10n_in_round_value(line.balance * sign),
         }
-        if tax_details_by_code.get("igst_rate"):
-            line_details.update({"igstRate": self._l10n_in_round_value(tax_details_by_code["igst_rate"])})
+        if tax_details_by_code.get("igst_rate") or (line.move_id.l10n_in_state_id.l10n_in_tin != line.company_id.state_id.l10n_in_tin and not tax_details_by_code.get("igst_rate")):
```

Comment:
The last 'and not' does not seem to make any sense (but the first get does)
```suggestion
        if tax_details_by_code.get("igst_rate") or (line.move_id.l10n_in_state_id.l10n_in_tin != line.company_id.state_id.l10n_in_tin):
```

Reply:
done